powertools-large-messages/src/main/java/software/amazon/lambda/powertools/largemessages/internal/LargeSQSMessageProcessor.java

```diff
@@ -0,0 +1,160 @@
+import java.nio.charset.StandardCharsets;
+import java.security.MessageDigest;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+public class LargeSQSMessageProcessor extends LargeMessageProcessor<SQSMessage> {
+
+    private static final Logger LOG = LoggerFactory.getLogger(LargeSQSMessageProcessor.class);
+    private static final String LEGACY_RESERVED_ATTRIBUTE_NAME = "SQSLargePayloadSize";
+    private static final int INTEGER_SIZE_IN_BYTES = 4;
+    private static final byte STRING_TYPE_FIELD_INDEX = 1;
+    private static final byte BINARY_TYPE_FIELD_INDEX = 2;
+    private static final byte STRING_LIST_TYPE_FIELD_INDEX = 3;
+    private static final byte BINARY_LIST_TYPE_FIELD_INDEX = 4;
+
+    @Override
+    protected String getMessageId(SQSMessage message) {
+        return message.getMessageId();
+    }
+
+    @Override
+    protected String getMessageContent(SQSMessage message) {
+        return message.getBody();
+    }
+
+    @Override
+    protected void updateMessageContent(SQSMessage message, String messageContent) {
+        message.setBody(messageContent);
+        // we update the MD5 digest so it doesn't look tempered
+        message.setMd5OfBody(calculateMessageBodyMd5(messageContent).orElse(message.getMd5OfBody()));
+    }
+
+    @Override
+    protected boolean isLargeMessage(SQSMessage message) {
+        Map<String, MessageAttribute> msgAttributes = message.getMessageAttributes();
+        return msgAttributes != null && (msgAttributes.containsKey(RESERVED_ATTRIBUTE_NAME) || msgAttributes.containsKey(LEGACY_RESERVED_ATTRIBUTE_NAME));
+    }
+
+    @Override
+    protected void removeLargeMessageAttributes(SQSMessage message) {
+        // message.getMessageAttributes() does not support remove operation, copy to new map
+        Map<String, MessageAttribute> newAttributes = new HashMap<>(message.getMessageAttributes());
+        newAttributes.remove(RESERVED_ATTRIBUTE_NAME);
+        newAttributes.remove(LEGACY_RESERVED_ATTRIBUTE_NAME);
+        message.setMessageAttributes(newAttributes);
+        // we update the MD5 digest so it doesn't look tempered
+        message.setMd5OfMessageAttributes(calculateMessageAttributesMd5(newAttributes).orElse(message.getMd5OfMessageAttributes()));
+    }
+
+    /**
+     * Compute the MD5 of the message body.<br/>
+     * Inspired from {@code software.amazon.awssdk.services.sqs.internal.MessageMD5ChecksumInterceptor}.<br/>
+     * package protected for testing purpose.
+     *
+     * @param messageBody body of the SQS Message
+     * @return the MD5 digest of the SQS Message body (or empty in case of error)
+     */
+    static Optional<String> calculateMessageBodyMd5(String messageBody) {
+        byte[] expectedMd5;
+        try {
+            expectedMd5 = Md5Utils.computeMD5Hash(messageBody.getBytes(StandardCharsets.UTF_8));
+        } catch (Exception e) {
```

Comment:
In what cases might this fail? I'm not sure if silently setting a null digest (and letting the client of powertools maybe have to debug into here to see what's happening) is better than throwing here 🧐

Reply:
It can throw an exception, we must deal with it...  I'm not throwing anything, just returning an empty string

Reply:
What happens when the user ends up with an empty MD5 on their message? Isn't it better to wrap in runtime and throw? This seems _exceptional_ - the user probably wants to see a failure, rather than e.g. comparing body to hash and gettting a mismatch -- ?

Reply:
It won't be empty, because if it's empty I put the original MD5. But I agree MD5 won't match if he tries to check (but will he be able to actually check if MD5 is not working for us, there are chances it won't for him..)
I didn't want to raise a exception as I'm pretty sure not so many people are verifying integrity of the messages/attributes...